### ISSUE TITLE
HiDPI scaling is not respected for menu icons and padding

### ISSUE THREAD

**Issue by paulmcauley** [Zren/material-decoration#39]
I have a HiDPI display at 250% scaling.

Here is a screenshot of how the Material menu looks:
![konsole_Material_menu](https://user-images.githubusercontent.com/6924479/111224589-91c47900-85d6-11eb-9455-52b8fc417471.png)

Here is a screenshot of how a normal menu looks:
![konsole_normal_menu](https://user-images.githubusercontent.com/6924479/111224630-a3a61c00-85d6-11eb-9e4e-5a684b853e46.png)

As you can see, the menu's icon size and padding are too small when scaling is used.

**Comment by Zren** [comment]
Was planning to refactor the `TextButton::horzPadding` from a hardcoded `4px` anyways, so I just made it use `smallSpacing`.

As for why the `QMenu` doesn't scale, I'm not sure.

**Comment by Zren** [comment]
Hmm, if I uncomment the code to display the normal "Application Menu" icon button that tells KWin to display the menu, it draws the entire menu at 100%. Even the text is at 100% scaling.

![2021-03-15___19-17-57](https://user-images.githubusercontent.com/416367/111233600-b400cb80-85c3-11eb-89c6-c5a3c94f45a6.png)

Even breeze seems to have this bug. Can you confirm this @paulmcauley?

![2021-03-15___19-20-14](https://user-images.githubusercontent.com/416367/111233603-b6632580-85c3-11eb-81c4-896813711610.png)

**Comment by paulmcauley** [comment]
No, I don't have the problem with a normal application menu in Breeze (Plasma 5.21.2). Here is what 250% with a normal application menu looks like for me: 
![breeze_application_menu_250](https://user-images.githubusercontent.com/6924479/111240744-b3802900-85f3-11eb-807d-f7b0a49e85f8.png)

**Comment by SeptemberHX** [comment]
It is possible to fix the small icon with QProxyStyle.

```cpp
    int pixelMetric(PixelMetric metric, const QStyleOption *option, const QWidget *widget) const override {
        qCDebug(category) << metric;
        if (metric == QStyle::PM_ButtonIconSize) {
            return QProxyStyle::pixelMetric(metric, option, widget) * pixelRatio;
        }
        return QProxyStyle::pixelMetric(metric, option, widget);
    }
```

Then `appMenu->setStyle(xxx)` should works.
But I don't know how to get the pixelRatio in a decoration.
